papermill/iorw.py

```diff
@@ -88,6 +88,12 @@ def listdir(cls, path):
 
     @classmethod
     def write(cls, buf, path):
+        dir = (path.split("/"))[:-1]
```

Comment:
I am not quite sure what the next few lines do but I think they try and find the directory name in the path. I'd recommend using https://docs.python.org/3/library/os.path.html#os.path.dirname or other tools from that module to do this.

By using an existing tool from the standard library you make it easier for others to read your code and we are less likely to have a bug in the code that manipulates the path.

Reply:
@betatim Sure. Will make the change as you suggested